support/cas-server-support-saml-idp-discovery/src/main/java/org/apereo/cas/services/DefaultSamlIdentityProviderDiscoveryFeedService.java

```diff
@@ -0,0 +1,101 @@
+package org.apereo.cas.services;
+
+import org.apereo.cas.configuration.CasConfigurationProperties;
+import org.apereo.cas.entity.SamlIdentityProviderEntity;
+import org.apereo.cas.entity.SamlIdentityProviderEntityParser;
+import org.apereo.cas.validation.DelegatedAuthenticationAccessStrategyHelper;
+import org.apereo.cas.web.DelegatedClientIdentityProviderConfiguration;
+import org.apereo.cas.web.DelegatedClientIdentityProviderConfigurationFactory;
+import org.apereo.cas.web.support.ArgumentExtractor;
+
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.util.InitializableObject;
+import org.pac4j.jee.context.JEEContext;
+import org.pac4j.saml.client.SAML2Client;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This is {@link DefaultSamlIdentityProviderDiscoveryFeedService}.
+ *
+ * @author Misagh Moayyed
```

Comment:
You will need to correct this.

Reply:
Sorry, is there a typo in the class name I'm missing?

Reply:
Hmmm. Still don't see the problem with this. If it is the author seemed fairest to keep you as the other as it is 90% just your code moved?